pkg/services/ngalert/api/authorization_test.go

```diff
@@ -47,7 +47,7 @@ func TestAuthorize(t *testing.T) {
 		}
 		paths[p] = methods
 	}
-	require.Len(t, paths, 30)
+	require.Len(t, paths, 32)
```

Comment:
Ah, this strikes again :)

Totally not a thing for this PR, but IMO this assert should go away. The point of this test is to verify that each route has an access control, no matter how many routes there are. No point in asserting the number of routes.

Reply:
Yes, that's super annoying :-D

Reply:
Playing devil's advocate here (mostly because I requested this test to be added 😆 ) - How do you know that this is asserting on all routes as opposed to none?

Perhaps, I come from a world where I like to exercise caution. If I change something somewhere, a test should tell me _what_ changed.

Reply:
Initially, I felt it was unusual to assert against declarative code. If a file contains just the definition of something, why also define it in tests and assert that the definitions match?

> How do you know that this is asserting on all routes as opposed to none?

I guess I'll slightly flip this question around - in what situation would this assertion ever catch a bug in Grafana? The Swagger doc is generated with `go-swagger` and then the tests uses `go-swagger` to get the list of routes from the doc. Would `go-swagger` itself have to have a bug in order for that to happen?

> If I change something somewhere, a test should tell me what changed.

The compiler will also catch this. If I a add route and run the Swagger generator, Grafana will fail to compile until I go and add an implementation of that route. It's not possible to change the number of routes and have no developer feedback whatsoever.

I suppose my underlying opinion is that if we have `n` tests with their own definition of something that's declarative, each change to that definition requires `n` places in the code to be updated. On top of that, such tests don't buy us much value, because the only way to break them is to change the declarative definition of something, which in most cases was done intentionally in the first place.